enterprise/backend/src/metabase_enterprise/sso/api/saml.clj

```diff
@@ -0,0 +1,32 @@
+(ns metabase-enterprise.sso.api.saml
+  "`/auth/saml` endpoints"
+  (:require
+   [compojure.core :refer [PUT]]
+   [metabase.api.common :as api]
+   [metabase.models.setting :as setting]
+   [metabase.util.schema :as su]
+   [saml20-clj.core :as saml]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+#_{:clj-kondo/ignore [:deprecated-var]}
+(api/defendpoint-schema PUT "/settings"
+  "Update SAML related settings. You must be a superuser or have `setting` permission to do this."
+  [:as {settings :body}]
+  {settings su/Map}
+  (api/check-superuser)
+  (let [filename (:saml-keystore-path settings)
+        password (:saml-keystore-password settings)
+        alias (:saml-keystore-alias settings)]
+    (if (or (every? empty? [filename password alias])
+            (saml/has-private-key? {:filename filename
+                                    :password password
+                                    :alias    alias}))
+      (t2/with-transaction [_conn]
+        (setting/set-many! settings))
+      ;; test failed, return result message
+      {:status 500
```

Comment:
Should the ldap response code also be 400?

Reply:
Oh hmm. I think in that case 500 makes more sense because it's testing a connection to a remote directory server, not just doing a validation of the values being set. So the connection test could fail because the server is down or something which wouldn't be an issue with the request.